src/interpreter.ts

```diff
@@ -1096,30 +1097,60 @@ export class Interpreter {
                         ast.args[0]!.loc,
                     );
 
-                    if (str.length > 255) {
-                        throwErrorConstEval(
-                            `hex string is too long, expected up to 255 characters, got ${str.length}`,
-                            ast.loc,
-                        );
-                    }
-                    if (!/^[0-9a-fA-F]*$/.test(str)) {
+                    if (!/^[0-9a-fA-F]*_?$/.test(str)) {
                         throwErrorConstEval(
                             `invalid hex string: ${str}`,
                             ast.loc,
                         );
                     }
 
-                    try {
-                        return beginCell()
-                            .storeBuffer(Buffer.from(str, "hex"))
-                            .endCell()
-                            .asSlice();
-                    } catch (_) {
+                    // Remove underscores from the hex string
+                    const hex = str.replace("_", "");
+                    const paddedHex = hex.length % 2 === 0 ? hex : "0" + hex;
```

Comment:
We need a test that justifies this `paddedHex` thing. AFAIU, if the padding is removed the tests won't catch it.

Reply:
there are multiple test cases with an odd number of digits already: `0_`, `1_`, `7`

Reply:
got it, thanks